### PR TITLE
add note about required_pkgs() in extension packages

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**

### CODE REVIEW

File: content/learn/develop/recipes/index.Rmarkdown [hunk 1]
```diff
@@ -380,6 +380,8 @@ required_pkgs.step_hypothetical <- function(x, ...) {
 
 In this example, `myrecipespkg` is the package where the step resides (if it is in a package).
 
+{{% note %}} If you are writing steps for an extension package, then every step should have a `required_pkgs()` method. It must return the package name, in addition to the required dependencies. {{%/ note %}}
```
Comment (simonpcouch) [inline content/learn/develop/recipes/index.Rmarkdown]:
```suggestion
{{% note %}} If you are writing steps for an extension package, then every step should have a `required_pkgs()` method. {{%/ note %}}
```

I think you've described the output format of these methods well in the lines preceding. It looks to me like the clarification here is that this method is _required_, and if that's the case, I'd include only that information in the note.

**Review by simonpcouch — COMMENTED** [review]
I don't know how these work, which may make me the intended audience, but left one comment!

Also, line 373 partially reads "There is an S3 method that can be used to declare what packages should be loaded when using the step." Should "can be used" read "should be used"? If a step requires no additional dependencies, does it need this method?

**Comment by EmilHvitfeldt** [comment]
You need to write one if
- You use an package inside your step
- the step is being put in an extension package

So steps that you write in your .R script itself doesn't need them, or steps in {recipes}. Unless of cause they use a package to do the calculation. So it is a little hairbally

File: content/learn/develop/recipes/index.Rmarkdown (hunk 1 above)
Comment (EmilHvitfeldt) [reply content/learn/develop/recipes/index.Rmarkdown]: That is a good note!

**Review by simonpcouch — APPROVED** [review]
Gotcha! Okay, looks good then.🙂

**Pull request merged**